table/block_based_table_reader.h

```diff
@@ -705,6 +710,8 @@ class BlockBasedTableIterator : public InternalIteratorBase<TValue> {
   TBlockIter block_iter_;
   bool block_iter_points_to_real_block_;
   bool is_out_of_bound_ = false;
+  // Whether current data block being fully within iterate upper bound.
+  bool data_block_within_upper_bound_ = true;
```

Comment:
Should be default false to be safe?

Reply:
Neither one is safer to me:
1. when used to check if next block needs to be read, setting to true is safer, it allow the next block to be read.
2. when used to check whether upper bound needs to be checked, setting to false is safer, it force upper bound to be checked.
The initial value shouldn't be used anyway, but I don't know a way to assert that.

Reply:
Then it's better to find why it failed in the first place then. It introduces a potential bug. When you try to reproduce, which flavor did you run? Did you run ASAN?

Reply:
btw, I don't understand 1. If the value is false, will we skip the next block? Why?

Reply:
if the current block is not completely within upper bound, the next block will be completely out of upper bound, so it's safe to skip.